### ISSUE TITLE
Failing tests

### ISSUE THREAD

**Issue by mfittere** [SixTrack/SixTrack#180]
Test report for the wire:
- gfortran: all test pass
- ifort: thck6ddynk fails (prob1/prob3 untested)
- nagfor: thck6ddynk fails (prob1/prob3 untested)

for all cr version the following tests seem to fail:
wire_thck4d wire_thck6d elensidealthck4d elensidealthck6d thick4 thick6dblocks thick6ddynk
in addition:
- gfortran cr: lostnumxv diff in fort.90
- ifort cr: lostnumxv diff in fort.90
- nagfor cr: crashes for frs60

**Comment by kyrsjo** [comment]
lostnumxv: #169

**Comment by JamesMolson** [comment]
On lxplus, using the master branch as is right now:

ifort + thick6ddynk:
```
SixTrack_4545_crlibm_fast_tilt_cmake_ifort_static_64bit 
$ ctest -R thick6ddynk
Test project /tmp/jamolson
    Start 37: thick6ddynk
1/1 Test #37: thick6ddynk ......................   Passed    0.33 sec

100% tests passed, 0 tests failed out of 1

Label Time Summary:
fast    =   0.33 sec (1 test)

Total Test time (real) =   0.35 sec
```

nagfor + thick6ddynk
```
Linking Fortran executable SixTrack_4545_crlibm_fast_tilt_cmake_nagfor_static_64bit
NAG Fortran Compiler Release 6.0(Hibiya) Build 1064
[100%] Built target SixTrack
Test project /tmp/jamolson/nagfor
    Start 37: thick6ddynk
1/1 Test #37: thick6ddynk ......................   Passed    0.37 sec

100% tests passed, 0 tests failed out of 1

Label Time Summary:
fast    =   0.37 sec (1 test)

Total Test time (real) =   0.46 sec
```

nagfor + CR tests
[100%] Linking Fortran executable SixTrack_4545_cr_crlibm_fast_tilt_cmake_nagfor_static_64bit
```
$ ctest -R "elensidealthck4d|elensidealthck6d|thick4|thick6dblocks|thick6ddynk|frs60" -j6
Test project /tmp/jamolson/nagfor
    Start 34: elensidealthck4d
    Start 20: frs60
    Start 31: thick4
    Start 32: thick6dblocks
    Start 35: elensidealthck6d
    Start 37: thick6ddynk
1/6 Test #32: thick6dblocks ....................   Passed    2.02 sec
2/6 Test #35: elensidealthck6d .................   Passed    2.32 sec
3/6 Test #37: thick6ddynk ......................   Passed    2.33 sec
4/6 Test #34: elensidealthck4d .................   Passed    3.15 sec
5/6 Test #31: thick4 ...........................   Passed   12.17 sec
6/6 Test #20: frs60 ............................   Passed  1299.97 sec

100% tests passed, 0 tests failed out of 6

Label Time Summary:
fast      =  21.99 sec (5 tests)
medium    = 1299.97 sec (1 test)

Total Test time (real) = 1299.98 sec
```


And finally, just to be sure, as mentioned above :
```
$ ctest -R lostnumxv
Test project /tmp/jamolson/nagfor
    Start 25: lostnumxv
1/1 Test #25: lostnumxv ........................***Failed    3.09 sec

0% tests passed, 1 tests failed out of 1

Label Time Summary:
fast    =   3.09 sec (1 test)

Total Test time (real) =   3.14 sec

The following tests FAILED:
	 25 - lostnumxv (Failed)
Errors while running CTest
```


I'll merge in the current wire branch and see what happens.

**Comment by JamesMolson** [comment]
Normal wire:
```
Test project /tmp/jamolson/nagfor
    Start 38: wire_thck4d
    Start 39: wire_thck6d
    Start 40: wire_thin4d
    Start 41: wire_thin6d
    Start 42: wire_thin6d_ibeco0
1/5 Test #38: wire_thck4d ......................   Passed    8.51 sec
2/5 Test #40: wire_thin4d ......................   Passed    8.60 sec
3/5 Test #39: wire_thck6d ......................   Passed    9.93 sec
4/5 Test #41: wire_thin6d ......................   Passed   22.07 sec
5/5 Test #42: wire_thin6d_ibeco0 ...............   Passed   22.79 sec

100% tests passed, 0 tests failed out of 5

Label Time Summary:
fast    =  71.90 sec (5 tests)

Total Test time (real) =  23.17 sec
```

wire + CR:

```
[100%] Linking Fortran executable SixTrack_4545_cr_crlibm_fast_tilt_cmake_nagfor_static_64bit
NAG Fortran Compiler Release 6.0(Hibiya) Build 1064
[100%] Built target SixTrack
jamolson@lxplus104.cern.ch:[nagfor]$ ctest -R wire -j5
Test project /tmp/jamolson/nagfor
    Start 38: wire_thck4d
    Start 39: wire_thck6d
    Start 40: wire_thin4d
    Start 41: wire_thin6d
    Start 42: wire_thin6d_ibeco0
1/5 Test #40: wire_thin4d ......................   Passed    7.14 sec
2/5 Test #38: wire_thck4d ......................   Passed   13.27 sec
3/5 Test #39: wire_thck6d ......................   Passed   14.07 sec
4/5 Test #41: wire_thin6d ......................   Passed   29.11 sec
5/5 Test #42: wire_thin6d_ibeco0 ...............   Passed   30.08 sec

100% tests passed, 0 tests failed out of 5

Label Time Summary:
fast    =  93.67 sec (5 tests)

Total Test time (real) =  30.09 sec

```

**Comment by kyrsjo** [comment]
Regarding Miriam's report:

Still to test for nagfor cr: `frs javier lostevery lost lostnumxv lostnotilt notilt s316 tilt fma`
Everything was compiled with make_six

**Comment by kyrsjo** [comment]
Something is wrong with prob1/prob3 when running with CR.
Ongoing tests (this post will be updated):
- Version 4.5.29 ( https://github.com/SixTrack/SixTrack/commit/62cf8da92d3f1f3711cec165d9c9df500063b513 ), built as `make_six gfortran -cernlib cr` - RUNNING

**Comment by kyrsjo** [comment]
Uh - problem found. Thje Sixin.zip files still has the RIPP block. Will update and make a PR. #182 would be nice now...

**Comment by kyrsjo** [comment]
See #190 for Sixin.zip update!

**Comment by JamesMolson** [comment]
@kyrsjo 
Please have a look at the following tests also:

thick6dsingles - just a fort.6
thick6dblocks + dynk_globalvars + elensidealthck6d - fort.10.canonical exists but has zero size.
Should thick6ddynk have a fort.10 ?

**Comment by kyrsjo** [comment]
* thick6dsingles : The fort.6 indicates that this test fails miserably. Apparently known to not work: fa7e60f3edeb09eeb76b1a3edff6a4fb1a543bf8
This should probably be removed from CTest (commented out). I remember some discussion about this test, apparently the fort.2 isn't really valid. @rdemaria may remember why?

* thick6dblocks + dynk_globalvars + elensidealthck6d: None of these have a `POST` block in their fort.3, so there is no post-processing and thus no (or empty) fort.10. However the CR version always produces some kind of fort.3... This is also true for `exact` (empty fort.10) and thick6ddynk (no fort.10). We should probably find one standard (either empty or no fort.10) and stick to it.

**Comment by kyrsjo** [comment]
I just ran the tests (`ctest -j3`) on Linux with (gfortran and ifort) x (CR and normal), master branch. Everything except lostnumxv-with-CR passed (prob1/prob3 still running). That means 35/37 tests have ran.

We don't have nagfor at the moment, something is wrong with the license server... Waiting for IT to fix it...

The following tests have NOT ran:
 * bnl : Only for BNL version
 * elensidealthin6d_DYNK : Disabled, missing canonicals - see #189 
 * javier_bignpart: Only for BIGNPART and HUGENPART version
 * thick6dsingles : Broken test. 

Starting mac tests (skipping prob1/prob3) now, will do windows tests as soon as #191 is fixed.

**Comment by kyrsjo** [comment]
Regarding the empty fort.10.canonicals: I think if fort.3 does NOT contain a POST, there should be no fort.10.canonical. The reason is that the CR version will ALWAYS produce a big fort.10 of mostly zeros, while the normal version will just produce an empty one.

In check_10, I think this is handled by the script checking if the fort.10 is non-existent OR empty, and in that case it skips the comparison. 

We should therefore delete the files `thick6dblocks/fort.10.canonical`, `dynk_globalvars/fort.10.canonical`, `elensidealthck6d/fort.10.canonical`, `exact/fort.10.canonical` + make the neccessary adjustments to `SixTest/CMakeLists.txt`. I'll make the PR.

**Comment by JamesMolson** [comment]
The 2 DA tests also do a write(*,*) and therefore the output varies on each compiler.

**Comment by JamesMolson** [comment]
Looking at the output of some of the tests, we have situations where there are some overflows of print statements.
These happen in all the "lost" tests and also in "s316" for me

**Comment by kyrsjo** [comment]
Just noticed (while testing for PR #263) that javier_bignpart fails.

**Comment by kyrsjo** [comment]
Fixed by #287
